### PR TITLE
Use oc adm instead of oadm which might not exist in various installations.

### PR DESCRIPTION
**Pull request by adelton**
Fixes https://github.com/openshift/openshift-docs/issues/1845.

### CODE REVIEW

**Comment by enj** [comment]
@smarterclayton do we have a preference for what binary someone uses?  Also, I do not know if we always have `oc` either, so I cannot tell if this really helps anything.

[test]

**Comment by adelton** [comment]
[test]

**Comment by enj** [comment]
@adelton that is not a flake, you incorrectly changed BASH completions.

**Comment by adelton** [comment]
Nod, I'll have to investigate some more. I don't see what exactly is failing, so I was not able to reproduce it locally yet.

**Comment by adelton** [comment]
Attempting to revert the `oc adm completion` changes to see if it improves the tests -> 3acd2a57bf30a1438cd1cf30aa7e83d45c356a8b.

**Comment by adelton** [comment]
I have removed any `oc adm completion` changes but the tests failed again. @enj, do you know what the failing tests do and how to reproduce locally or how to investigate in that Jenkins environment?

**Comment by smarterclayton** [comment]
Re: preference, we prefer oc adm.  oadm is just a convenience

File: test/cmd/basicresources.sh [hunk 1]
```diff
@@ -58,10 +58,10 @@ os::cmd::expect_success_and_text 'openshift kubectl get pods --loglevel=7  2>&1
 os::cmd::expect_success_and_text 'openshift kubectl get dc --loglevel=7  2>&1 | grep -A4 "deploymentconfig" | grep User-Agent' "openshift/${kube_git_regex} .* kubernetes/"
 # example User-Agent: oadm/v1.2.0 (linux/amd64) kubernetes/bc4550d
 # this is probably broken and should be `oadm/<oc version>... openshift/...`
-os::cmd::expect_success_and_text 'oadm policy reconcile-sccs --loglevel=7  2>&1 | grep -A4 "securitycontextconstraints" | grep User-Agent' "oadm/${kube_git_regex} .* kubernetes/"
+os::cmd::expect_success_and_text 'oc adm policy reconcile-sccs --loglevel=7  2>&1 | grep -A4 "securitycontextconstraints" | grep User-Agent' "oadm/${kube_git_regex} .* kubernetes/"
 # example User-Agent: oadm/v1.1.3 (linux/amd64) openshift/b348c2f
 # TODO: figure out why this is reporting openshift and not kubernetes
-os::cmd::expect_success_and_text 'oadm policy who-can get pods --loglevel=7  2>&1 | grep -A4 "localresourceaccessreviews" | grep User-Agent' "oadm/${os_git_regex} .* openshift/"
+os::cmd::expect_success_and_text 'oc adm policy who-can get pods --loglevel=7  2>&1 | grep -A4 "localresourceaccessreviews" | grep User-Agent' "oadm/${os_git_regex} .* openshift/"
```
Comment (Miciah) [inline test/cmd/basicresources.sh]: The user agent needs to be changed from "oadm" to "oc".

File: hack/lib/start.sh [hunk 2]
```diff
@@ -607,11 +607,11 @@ function os::start::internal::print_server_info() {
 #  None
 function os::start::router() {
 	os::log::debug "Installing the router"
-	oadm policy add-scc-to-user privileged --serviceaccount='router' --config="${ADMIN_KUBECONFIG}"
+	oc adm policy add-scc-to-user privileged --serviceaccount='router' --config="${ADMIN_KUBECONFIG}"
 	# Create a TLS certificate for the router
 	if [[ -n "${CREATE_ROUTER_CERT:-}" ]]; then
 		os::log::debug "Generating router TLS certificate"
-		oadm ca create-server-cert --hostnames="*.${API_HOST}.nip.io"          \
+		oc adm ca create-server-cert --hostnames="*.${API_HOST}.nip.io"          \
```
Comment (Miciah) [inline hack/lib/start.sh]: Need to adjust whitespace to avoid unsightly misalignment of the backslashes.

File: test/cmd/basicresources.sh (hunk 1 above)
Comment (adelton) [inline test/cmd/basicresources.sh]: Fixed in 32c6e2c8dc02de4a55c98b1830fb5f1d687c987f.

File: hack/lib/start.sh (hunk 2 above)
Comment (adelton) [inline hack/lib/start.sh]: Fixed in 32c6e2c8dc02de4a55c98b1830fb5f1d687c987f.

**Comment by adelton** [comment]
Rebased on master -> 32c6e2c8dc02de4a55c98b1830fb5f1d687c987f.

**Comment by adelton** [comment]
Fixed the failing tests -> c7913ab377cc6d78a9ba90abf1b092930c84c4f4.

**Comment by adelton** [comment]
> we prefer oc adm. oadm is just a convenience

Thanks, @smarterclayton. This pull request was now fixed to have the checks passing. It modifies files all across the place -- who would be the best person to review and potentially give the green light?

**Comment by adelton** [comment]
Rebased on master to resolve conflicts -> 40c2161da77e5f55bf291a1d1cad39a2cacc77fd.

**Comment by adelton** [comment]
What does that needs-api-review label mean?

The "Submit Queue" check is yellow -- but I have no idea what is wrong there and the "Details" link does not provide any clue.

**Comment by smarterclayton** [comment]
Anything which changes the API gets tagged that way.

**Comment by adelton** [comment]
Rebased on master -> 95d065a505278d75a8d3d22a02893bf2e375af98.

**Comment by adelton** [comment]
Rebased on master -> ce7709e81b90e24aebfb5366001645a7e7d78fd8.

**Comment by adelton** [comment]
@eparis, what would be the best way to get this pull request (which touches things across the repository) reviewed?

**Pull request merged**